src/api/app/models/concerns/staging_project.rb

```diff
@@ -142,35 +142,26 @@ def cache_problems
   end
 
   def state
-    return :empty unless staged_requests.exists?
-    return :unacceptable if unacceptable_state?
-    return :review if review_state?
-    general_state
-  end
-
-  def unacceptable_state?
-    untracked_requests.present? || staged_requests.obsolete.present?
-  end
-
-  def review_state?
-    general_state == :acceptable && missing_reviews.present?
+    return :empty if staged_requests.blank?
+    return :unacceptable if untracked_requests.present? || staged_requests.obsolete.present?
+    return build_and_check_state if build_and_check_state
+    return :review if missing_reviews.present?
+    :acceptable
   end
 
-  def general_state
-    build_state == :acceptable ? check_state : build_state
+  def build_and_check_state
```

Comment:
[nitpick] Why not calling both methods directly instead of having a method to wrap them?

Reply:
Because doing it increased the complexity of the main method and Code Climate complained. 
We also thought about removing the two methods `build_state` and `check_state`and put all their code directly inside the wrapper. But we considered that, semantically, it's better like this.

Reply:
I am not sure if we should follow code climate advice in this case, I find this more complex to follow...

Reply:
@Ana06  We have changed it a bit. I think it's clearer now. Please have a look!